controlers/controlers.js

```diff
@@ -40,13 +41,24 @@ exports.getArticleById = (req, res, next) => {
 };
 
 exports.getArticles = (req, res, next) => {
-  fetchArticles()
-    .then((result) => {
-      res.status(200).send(result);
-    })
-    .catch((err) => {
-      next(err);
-    });
+  const { topic } = req.query;
+  if (topic) {
+    fetchArticlesByTopic(topic)
+      .then((articles) => {
+        res.status(200).send(articles);
+      })
+      .catch((err) => {
+        next(err);
+      });
+  } else {
+    fetchArticles()
+      .then((articles) => {
+        res.status(200).send(articles);
+      })
+      .catch((err) => {
+        next(err);
+      });
```

Comment:
I don't think you need two invocations / separate models for this endpoint - have a think about how you could do a check here and if there's no topic, still send it to the model :)